testbench/common.py

```diff
@@ -662,8 +697,11 @@ def streamer():
                     yield d[r:chunk_end]
                 if bytes_yield >= limit:
                     database.dequeue_next_instruction(test_id, method)
-                    if conn is not None:
-                        conn.close()
+                    # Accessing request object in unit test will give "working out of request context" exception. Ignoring these exception.
+                    try:
+                        _make_closer(request)
```

Comment:
`_make_closer(request)` returns a function that should be called to have any effect?

Reply:
> It seems that some comments are still unaddressed, and the code may does not work as I would expect.

I've addressed all comments now. Please let me know if any of the change is not inline with what you have suggested,